ion-c-sys/src/lib.rs

```diff
@@ -306,6 +311,315 @@ mod test {
     }
 }
 
+/// Creates an operating context for decNum with maximum ranges primarily for conversions
+/// between `ION_DECIMAL` and `BigDecimal`.
+///
+/// This range is still below the capabilities of `BigDecimal`
+#[inline]
+fn make_context() -> decContext {
+    let mut ctx = decContext::default();
+    unsafe { decContextDefault(&mut ctx, DEC_INIT_DECQUAD as i32) };
+    ctx.digits = DEC_MAX_DIGITS as i32;
+    ctx.emax = DEC_MAX_EMAX as i32;
+    ctx.emin = DEC_MIN_EMIN as i32;
+    // make sure we don't pad decQuads
+    ctx.clamp = 0;
+
+    ctx
+}
+
+const DEC_QUAD_DIGITS: u64 = 37;
```

Comment:
I should look this up, but hopefully you have the answer in your RAM :) 
I thought the capacity for DIGITS is what we pass at compile time https://github.com/amzn/ion-rust/blob/90efa71bf3af14e4f4bd54f4be06fd0a3bfbcdcf/ion-c-sys/build.rs#L36

Why is it 37? 

And from lines 349 - 350 in this PR I am sure I am missing something that has to do with Ion C :)

Reply:
IIRC, this is to enable the `decQuad` type which are 128-bit decimals that have 37 digits of precision.  One can still use the `decNumber` type for decimal numbers that are beyond this.

Reply:
Ah, here it is in the `CMakeLists.txt` for Ion C:

```cmake
# NOTE: DECNUMDIGITS must be set across all compilation units to at least DECQUAD_Pmax (34), so that the value is
# guaranteed to be consistent between ionc and decNumber. This is required for conversions between decQuad and
# decNumber. This is NOT the limit on decimal precision; ION_DECIMAL can handle arbitrarily large precision.
add_definitions(-DDECNUMDIGITS=34)
```

[1]: https://github.com/amzn/ion-c/blob/1e911eb689a879427aa8842fe2ca7c78546aeed1/CMakeLists.txt#L22-L26

Reply:
I've updated the `build.rs` doc in [this commit](https://github.com/amzn/ion-rust/pull/81/commits/aeaa47606ea2c3d814a00c82f3958da8000cbc7f).

I've also re-thought what the heck I was doing there and fixed #79 with [this commit](https://github.com/amzn/ion-rust/pull/81/commits/a6ed8e093d7705f67983fafb1c492ce72e39f5db).

(All of the above are in this PR and reviewable separately)